### DISCUSSION TITLE
Add support for alternative / disabled taiko hit flying animation

### DISCUSSION THREAD

**Discussion by MikuOgiwara** [ppy/osu#23656] [Ideas]
Can the animation of the notes when they are hit be changed via a parameter between the new and the old or the notes once hited go in the life bar? 

and same thing fond the background for put the old method to display background in game ? 

thx <3

**Comment by MikuOgiwara** [comment]
any news ?

**Reply by peppy** [reply]
no

**Comment by Briesmas** [comment]
Bumping this as it will bother many taiko players.
Most of us are either used to:
- have the notes go directly into the life bar
- have no animation by disabling it with shift+tab

This is quite distracting for a lot of players who have been playing this way for multiple years so I hope this can be considered.

**Comment by Eltigant** [comment]
Bumping this as well.
An easy fix to this issue might be as simple as adding a setting:

-  Show _flying notes_ after hitting the notes. (On/Off/Only when UI is disabled/only when UI is enabled)

This setting would allow players to simply hide it while the UI is also hidden (like stable's "shift + tab"), as well as adding an option to hide those notes while having the UI on. Having notes moving in the area can be very distracting to people (myself including lol), but a lot of players wouldn't mind having the accuracy/hp bar showing.

**Reply by TheFearlessDeath500** [reply]
I think the flying notes wouldn't be _entirely_ disabled. Rather, it should be a thing that can be skin-related more than anything.
If a player can change the velocity and trajectory of a note using something like the skin editor on lazer, I'm sure it'll be more fun for other players to customize the game to their liking.

**Comment by PixeLInfinity** [comment]
Bumping this again as a player who primarily played Stable with HUD off for exactly this effect
taiko on lazer has improved a lot and features like this will hopefully get even more taiko players to give lazer a chance

**Comment by TrueMushReen** [comment]
Giving this a bump because I got used to stable's _flying notes_ and I feel like I have no idea what notes I have already hit during long patterns or stream maps. Hopefully the switch to change hit animation got implemented.

**Comment by TheFearlessDeath500** [comment]
Bumping yet again. This needs to be addressed since day one.
Either that or give an option to the player to change the direction of where the note fades, where it starts and where it ends. Probably more like an option in the in-game "skin editor" that's offered with lazer.
Heck, some players that I know still have this issue too (Eltigant as an example). This **NEEDS** to be fixed ASAP.

Also another thing: Depending on the current BPM of the map, this also changes how far the notes travels too. Play a map with a lot of notes on screen with a low SV and BPM, and you'll see what I mean.

**Reply by peppy** [reply]
If you are able to write code, you can MAKE this fixed ASAP 😉

**Reply by Issuko-Adam** [reply]
Impressive how you ask for feedback from taiko players to just reply this a year later!
you could have just considered the issue/remark
Keep up the good work peppy, but taiko playerbase will never make it through lazer.

**Reply by peppy** [reply]
Never say never.

Understand that we want to make changes like this but are spread thin. Bumping with bold expectations and aggressive wording like "addressed since day one" is not the way to bump and make us want to prioritise an issue.

We want to make this change and will, but I suggest that someone who is able to help jumps in and writes the implementation to help speed things up.

**Comment by cdwcgt** [comment]
https://github.com/ppy/osu/assets/34775378/3b812a0f-e9fa-4b24-a775-149d548a3bba

I wrote these for now, but I am not confident about the impact of lifetime on the judgment, but the test seems to be fine.

this implementation cannot adjust the flying position
I don't know if I should add curve markers or something like that in the skin editor, I've never done it before, so I left it as is.

Another point is hidden, notes after hidden will never fly, it can easily read config to let hidden consider that should note fly, but it little out of scope
